ui/app/AppLayouts/Profile/panels/ProfileShowcasePanel.qml

```diff
@@ -0,0 +1,350 @@
+                    showcaseModel.append(tmpObj)
+                }
+            }
+        }
+
+        contentItem: Item {
+            RowLayout {
+                anchors.centerIn: parent
+                spacing: visibilityDropAreaLocal.spacing
+                StatusIcon {
+                    width: 20
+                    height: 20
+                    icon: ProfileUtils.visibilityIcon(visibilityDropAreaLocal.showcaseVisibility)
+                    color: visibilityDropAreaLocal.icon.color
+                }
+                StatusBaseText {
+                    Layout.fillWidth: true
+                    font.pixelSize: 13
+                    font.weight: Font.Medium
+                    color: visibilityDropAreaLocal.icon.color
+                    text: visibilityDropAreaLocal.text
+                }
+            }
+        }
+    }
+
+    Component.onCompleted: showcaseModel.load()
+    Component.onDestruction: showcaseModel.save()
+
+    // NB temporary model until the backend knows the extra roles: "showcaseVisibility" and "order"
+    ListModel {
+        id: showcaseModel
+
+        function hasItem(itemId) {
+            for (let i = 0; i < count; i++) {
+                let item = get(i)
+                if (!!item && item[root.keyRole] === itemId)
+                    return true
+            }
+            return false
+        }
+
+        function save() {
+            var result = []
+            for (let i = 0; i < count; i++) {
+                let item = get(i)
+                result.push(item)
+            }
+            settings.setValue(root.settingsKey, JSON.stringify(result))
+        }
+
+        function load() {
+            const data = settings.value(root.settingsKey)
+            try {
+                const arr = JSON.parse(data)
+                for (const i in arr)
+                    showcaseModel.append(arr[i])
+            } catch (e) {
+                console.warn(e)
+            }
+        }
+    }
+
+    Settings {
+        id: settings
+        category: "Showcase"
+
+        function reset() {
+            showcaseModel.clear()
+            settings.setValue(root.settingsKey, "")
+            settings.sync()
+        }
+    }
+
+    QtObject {
+        id: d
+
+        readonly property int defaultDelegateHeight: 60
+        readonly property int contentSpacing: 12
+        readonly property int strokeMargin: 2
+    }
+
+    contentItem: ColumnLayout {
+        id: layout
+        spacing: d.contentSpacing
+
+        StatusBaseText {
+            Layout.fillWidth: true
+            text: qsTr("In showcase")
+            color: Theme.palette.baseColor1
+            font.pixelSize: 13
+            font.weight: Font.Medium
+        }
+
+        StatusListView {
+            id: showcaseItemsListView
+            Layout.fillWidth: true
+            Layout.preferredHeight: count ? contentHeight : targetDropArea.height
+            model: showcaseModel
+            interactive: false
```

Comment:
Yup... so the slowness is because of recalculating the overall height of the panel. And that has issues in itself too. Working on a slightly different approach

Reply:
Pushed a fix for these as well, let's discuss that later, needs some explanation and maybe some further discussions :)